gremlin-go/driver/graph.go

```diff
@@ -120,3 +126,59 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, fmt.Errorf("path does not contain a label of '%s'", key)
 	}
 }
+
+// Set describes the necessary methods that need to be implemented for Gremlin-Go to recognize for use as a Gremlin Set.
+type Set interface {
+	// ToSlice is the only method that needs to be implemented in order for a custom Set to operate properly.
+	// ToSlice must return a slice that contains all the elements of the underlying Set with no duplicates.
+	ToSlice() []interface{}
+}
+
+// SimpleSet is a basic implementation of a Set for use with Gremlin-Go.
+type SimpleSet struct {
+	objects []interface{}
+}
+
+func (s *SimpleSet) ToSlice() []interface{} {
+	return s.objects
+}
+
+// Add adds an item to the SimpleSet only if it is not already a part of it
+func (s *SimpleSet) Add(val interface{}) {
+	if !s.Contains(val) {
+		s.objects = append(s.objects, val)
+	}
+}
+
+// Remove is Shallow comparison operation that removes a value from the SimpleSet
+func (s *SimpleSet) Remove(val interface{}) {
+	for i, obj := range s.objects {
+		if val == obj {
+			// Deletion does not maintain ordering
+			s.objects[i] = s.objects[len(s.objects)-1]
+			s.objects = s.objects[:len(s.objects)-1]
+			break
+		}
+	}
+}
+
+// Contains checks if a value is contained in the SimpleSet or not.
+// Items are considered already contained in the set according to Golang equality operator rules.
+func (s *SimpleSet) Contains(val interface{}) bool {
+	for _, entry := range s.ToSlice() {
+		if val == entry {
+			return true
+		}
+	}
+	return false
+}
+
+// NewSimpleSet constructs a new SimpleSet
```

Comment:
`NewSimpleSet creates a new SimpleSet containing all passed in args with duplicates excluded according to Golang equality operator rules.`